compiler/aarch64/codegen/OMRMemoryReference.cpp

```diff
@@ -86,17 +88,136 @@ bool OMR::ARM64::MemoryReference::useIndexedForm()
+   TR::RealRegister *index = self()->getIndexRegister() ? toRealRegister(self()->getIndexRegister()) : NULL;
+   TR::RealRegister *target = toRealRegister(currentInstruction->getMemoryDataRegister());
+
+   if (self()->getUnresolvedSnippet())
+      {
+      TR_ASSERT(false, "Not implemented yet.");
+      }
+   else
+      {
+      int32_t displacement = self()->getOffset();
+
+      TR::InstOpCode op = currentInstruction->getOpCode();
+      uint32_t enc = (uint32_t)op.getOpCodeBinaryEncoding();
+
+      if (index)
+         {
+         TR_ASSERT(displacement == 0, "Non-zero offset with index register.");
+
+         if (isRegisterOffsetInstruction(enc))
+            {
+            base->setRegisterFieldRN(wcursor);
+            index->setRegisterFieldRM(wcursor);
+            target->setRegisterFieldRT(wcursor);
+
+            if (self()->getScale() != 0)
+               {
+               TR_ASSERT(false, "Not implemented yet.");
+               }
+
+            cursor += ARM64_INSTRUCTION_LENGTH;
+            }
+         else
+            {
+            TR_ASSERT(false, "Unsupported instruction type.");
+            }
+         }
+      else
+         {
+         /* no index register */
+         base->setRegisterFieldRN(wcursor);
+         target->setRegisterFieldRT(wcursor);
+
+         if (isImm9OffsetInstruction(enc))
+            {
+            if (constantIsImmed9(displacement))
+               {
+               *wcursor |= (displacement & 0x1ff) << 12; /* imm9 */
+               cursor += ARM64_INSTRUCTION_LENGTH;
+               }
+            else
+               {
+               /* Need additional instructions for large offset */
+               TR_ASSERT(false, "Not implemented yet.");
+               }
+            }
+         else if (isImm12OffsetInstruction(enc))
+            {
+            int32_t size = (enc >> 30) & 3; /* b=0, h=1, w=2, x=3 */
+            int32_t shifted = displacement >> size;
+
+            if (size > 0)
+               {
+               TR_ASSERT((displacement & ((1 << size) - 1)) == 0, "Non-aligned offset in halfword memory access.");
+               }
+
+            if (constantIsUnsignedImmed12(shifted))
+               {
+               *wcursor |= (shifted & 0xfff) << 10; /* imm12 */
+               cursor += ARM64_INSTRUCTION_LENGTH;
+               }
+            else
+               {
+               /* Need additional instructions for large offset */
+               TR_ASSERT(false, "Not implemented yet.");
+               }
+            }
+         else
+            {
+            /* Register pair, literal, exclusive instructions to be supported */
+            TR_ASSERT(false, "Not implemented yet.");
+            }
+         }
+      }
 
    return cursor;
    }
 
 
 uint32_t OMR::ARM64::MemoryReference::estimateBinaryLength(TR::CodeGenerator& codeGen)
    {
-   TR_ASSERT(false, "Not implemented yet.");
```

Comment:
I suppose the question is why this function takes a `TR::CodeGenerator` parameter (passed by reference no less) when it's not even used or needed.  I'd say remove it until it's needed (unless you believe some of the unimplemented paths are going to need it, in which case I'd say pass in the pointer to be consistent with most of the other code generators).